### PR TITLE
hal: esp: pm: light sleep support

### PR DESCRIPTION
**Pull request by glaubermaroto**
Hal support for light sleep.

### CODE REVIEW

File: zephyr/esp32c3/CMakeLists.txt [hunk 1]
```diff
@@ -66,11 +59,14 @@ if(CONFIG_SOC_ESP32C3)
     ../../components/wpa_supplicant/esp_supplicant/include
     ../../components/wpa_supplicant/src/utils
     ../../components/wpa_supplicant/src/eap_peer
-
     ../../components/heap/include
     ../../components/mbedtls/port/include
     )
 
+  zephyr_include_directories(
```
Comment (sylvioalves) [inline zephyr/esp32c3/CMakeLists.txt:66]: This is duplicated.

File: zephyr/esp_shared/components/driver/gpio.c [hunk 2]
```diff
@@ -153,6 +153,13 @@ esp_err_t gpio_wakeup_enable(gpio_num_t gpio_num, gpio_int_type_t intr_type)
     return ret;
 }
 
+#if defined(__ZEPHYR__) && defined(CONFIG_PM)
+esp_err_t esp_gpio_wakeup_enable(gpio_num_t gpio_num, gpio_int_type_t intr_type)
+{
+	return gpio_wakeup_enable(gpio_num, intr_type);
```
Comment (sylvioalves) [inline zephyr/esp_shared/components/driver/gpio.c:159]: Identantion seems wrong. Please check.

File: zephyr/esp_shared/components/driver/rtc_io.c [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+static const char __attribute__((__unused__)) *RTCIO_TAG = "RTCIO";
+
+static int rtc_spinlock;
+
+#if SOC_RTCIO_INPUT_OUTPUT_SUPPORTED
+
+/*---------------------------------------------------------------
+                        RTC IO
+---------------------------------------------------------------*/
+esp_err_t rtc_gpio_init(gpio_num_t gpio_num)
```
Comment (sylvioalves) [inline zephyr/esp_shared/components/driver/rtc_io.c:26]: Same comments as in DAC driver.  
Can we work out the Macro logging system to meet Zephyr's?

Comment (glaubermaroto) [reply zephyr/esp_shared/components/driver/rtc_io.c:26]: Yes, it is possible but requires a wider test scope than this PR has.

Other components rely on the ESP macros logging system and could silently break, IMO, it should be addressed on another ticket.

File: zephyr/esp32c3/CMakeLists.txt (hunk 1 above)
Comment (glaubermaroto) [reply zephyr/esp32c3/CMakeLists.txt]: Fixed.

File: zephyr/esp_shared/components/driver/rtc_io.c (hunk 3 above)
Comment (sylvioalves) [reply zephyr/esp_shared/components/driver/rtc_io.c:26]: Solved in another PR.

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**